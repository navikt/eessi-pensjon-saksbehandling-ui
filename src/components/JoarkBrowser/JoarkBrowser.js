import React, { useState, useEffect } from 'react'
import PT from 'prop-types'
import _ from 'lodash'
import { connect, bindActionCreators } from 'store'
import * as joarkActions from 'actions/joark'
import * as uiActions from 'actions/ui'
import { getDisplayName } from 'utils/displayName'
import { NavFrontendSpinner } from 'components/Nav'
import File from 'components/File/File'
import TableSorter from 'components/TableSorter/TableSorter'

import './JoarkBrowser.css'

const mapStateToProps = (state) => {
  return {
    aktoerId: state.app.params.aktoerId,
    file: state.joark.file,
    list: state.joark.list,
    loadingJoarkList: state.loading.loadingJoarkList,
    loadingJoarkFile: state.loading.loadingJoarkFile,
    loadingJoarkPreviewFile: state.loading.loadingJoarkPreviewFile,
    previewFile: state.joark.previewFile
  }
}

const mapDispatchToProps = (dispatch) => {
  return { actions: bindActionCreators({ ...uiActions, ...joarkActions }, dispatch) }
}

export const JoarkBrowser = (props) => {
  const { actions, aktoerId, file, files, list, loadingJoarkList } = props
  const { loadingJoarkFile, loadingJoarkPreviewFile, onFilesChange, previewFile, t } = props
  const [_file, setFile] = useState(file)
  const [_files, setFiles] = useState(files)
  const [_previewFile, setPreviewFile] = useState(previewFile)
  const [mounted, setMounted] = useState(false)

  useEffect(() => {
    if (!mounted && list === undefined && !loadingJoarkList) {
      actions.listJoarkFiles(aktoerId)
    }
    setMounted(true)
  }, [mounted, list, loadingJoarkList, actions, aktoerId])

  useEffect(() => {
    if (file && (!_file ||
      (_file.dokumentInfoId !== file.dokumentInfoId || _file.content.base64 !== file.content.base64))) {
      setFile(file)
      const newFiles = _.cloneDeep(_files)
      newFiles.push(file)
      setFiles(newFiles)
      if (onFilesChange) {
        onFilesChange(newFiles)
      }
    }
  }, [file, _file, _files, onFilesChange])

  useEffect(() => {
    const onPreviewFile = (previewFile) => {
      actions.openModal({
        modalContent: (
          <div style={{ cursor: 'pointer' }} onClick={() => actions.closeModal()}>
            <File t={t} file={previewFile} width={400} height={600} />
          </div>
        )
      })
    }
    if (previewFile && (!_previewFile || _previewFile.journalpostId !== previewFile.journalpostId ||
     _previewFile.variantformat !== previewFile.variantformat)) {
      setPreviewFile(previewFile)
      onPreviewFile(previewFile)
    }
  }, [actions, previewFile, _previewFile, t])

  const onItemClicked = (clickedItem, clickedVariantFormat) => {
    const foundFile = _.find(files, (file) => {
      return file.journalpostId === clickedItem.raw.journalpostId &&
        file.variantformat === clickedVariantFormat &&
        file.content !== undefined
    })

    if (!foundFile) {
      actions.previewJoarkFile(clickedItem.raw, clickedVariantFormat)
    } else {
      setPreviewFile(foundFile)
    }
  }

  const onSelectedItemChange = (item, checked, variantformat) => {
    let newFiles = _.cloneDeep(_files)
    const found = _.find(newFiles, {
      dokumentInfoId: item.raw.dokumentInfoId,
      variantformat: variantformat
    })
    let changed = false

    if (!checked && found) {
      newFiles = _.reject(newFiles, {
        journalpostId: item.raw.journalpostId,
        variantformat: variantformat
      })
      changed = true
    }

    if (checked && !found) {
      newFiles.push({
        ...item.raw,
        variantformat: variantformat
      })
      changed = true
    }

    if (changed) {
      setFiles(newFiles)
      if (onFilesChange) {
        onFilesChange(newFiles)
      }
    }
  }

  const items = list ? list.map((file) => {
    return {
      raw: file,
      id: file.journalpostId,
      name: file.tittel,
      tema: file.tema,
      date: file.datoOpprettet,
      varianter: file.varianter.map(item => {
        return {
          variantformat: item.variantformat,
          filnavn: item.filnavn,
          label: item.variantformat + ' (' + item.filnavn + ')',
          selected: _.find(files, {
            dokumentInfoId: file.dokumentInfoId,
            variantformat: item.variantformat
          }) !== undefined,
          focused: _previewFile ? _previewFile.journalpostId === file.journalpostId && _previewFile.variantformat === item.variantformat : false
        }
      })
    }
  }) : []

  if (!mounted) {
    return null
  }

  if (loadingJoarkList) {
    return (
      <div>
        <NavFrontendSpinner type='XS' />
        <span className='pl-2'>{t('ui:loading')}</span>
      </div>
    )
  }

  return (
    <div className='c-joarkBrowser'>
      <TableSorter
        t={t}
        items={items}
        actions={actions}
        loadingJoarkFile={loadingJoarkFile}
        loadingJoarkPreviewFile={loadingJoarkPreviewFile}
        sort={{ column: 'name', order: 'desc' }}
        columns={{
          name: { name: t('ui:title'), filterText: '', defaultSortOrder: '' },
          tema: { name: t('ui:tema'), filterText: '', defaultSortOrder: '' },
          date: { name: t('ui:date'), filterText: '', defaultSortOrder: '' },
          varianter: { name: t('ui:variant'), filterText: '', defaultSortOrder: '' }
        }}
        onItemClicked={onItemClicked}
        onSelectedItemChange={onSelectedItemChange}
      />
    </div>
  )
}

JoarkBrowser.propTypes = {
  actions: PT.object.isRequired,
  aktoerId: PT.string,
  file: PT.object,
  files: PT.array,
  list: PT.array,
  loadingJoarkList: PT.bool,
  loadingJoarkFile: PT.bool,
  loadingJoarkPreviewFile: PT.bool,
  onFilesChange: PT.func.isRequired,
  previewFile: PT.object,
  t: PT.func.isRequired
}

const ConnectedJoarkbrowser = connect(mapStateToProps, mapDispatchToProps)(JoarkBrowser)
ConnectedJoarkbrowser.displayName = `Connect(${getDisplayName(JoarkBrowser)})`
export default ConnectedJoarkbrowser
