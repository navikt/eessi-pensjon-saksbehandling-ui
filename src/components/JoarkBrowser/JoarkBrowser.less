.c-joarkbrowser__subcell {

  display: flex;
  align-items: flex-start;
  padding-top: 0.25rem;
  justify-content: space-between;
  align-items: center;
  label {
    display: block !important;
  }
}
