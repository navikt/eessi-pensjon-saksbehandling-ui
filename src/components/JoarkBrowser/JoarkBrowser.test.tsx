import React from 'react'
import { JoarkBrowser, JoarkBrowserProps } from './JoarkBrowser'
import { mount, ReactWrapper } from 'enzyme'
import sampleJoark from 'resources/tests/sampleJoarkRaw'
import { JoarkDoc, JoarkFile, JoarkPoster } from 'declarations/joark'

describe('components/JoarkBrowser/JoarkBrowser', () => {
  let wrapper: ReactWrapper
  const files: Array<JoarkFile> = []
  sampleJoark.mockdata.data.dokumentoversiktBruker.journalposter.forEach((post: JoarkPoster) => {
    post.dokumenter.forEach((doc: JoarkDoc) => {
      files.push({
        tilleggsopplysninger: post.tilleggsopplysninger,
        journalpostId: post.journalpostId,
        tittel: doc.tittel,
        tema: post.tema,
        dokumentInfoId: doc.dokumentInfoId,
        datoOpprettet: new Date(Date.parse(post.datoOpprettet)),
        variant: doc.dokumentvarianter[0]
      })
    })
  })
  const initialMockProps: JoarkBrowserProps = {
    actions: {
      listJoarkFiles: jest.fn(),
      getPreviewJoarkFile: jest.fn()
    },
    aktoerId: '123',
    file: undefined,
    files: [],
    list: files,
    loadingJoarkList: false,
    loadingJoarkFile: false,
    loadingJoarkPreviewFile: false,
    onFilesChange: jest.fn(),
    mode: 'view',
    onPreviewFile: jest.fn(),
    previewFile: undefined,
    t: jest.fn(t => t)
  }

  beforeEach(() => {
    wrapper = mount(<JoarkBrowser {...initialMockProps} />)
  })

  afterEach(() => {
    wrapper.unmount()
  })

  it('Renders', () => {
    expect(wrapper.isEmptyRender()).toBeFalsy()
  })

  it('Render: loading', () => {
    wrapper = mount(<JoarkBrowser {...initialMockProps} loadingJoarkList />)
    expect(wrapper.find('WaitingPanel')).toBeTruthy()
  })

  it('Has proper HTML structure', () => {
    expect(wrapper.exists('.c-joarkBrowser')).toBeTruthy()
    expect(wrapper.exists('.c-tableSorter')).toBeTruthy()
  })

  it('UseEffect: list Joark files ', () => {
    // @ts-ignore
    wrapper = mount(<JoarkBrowser {...initialMockProps} list={undefined} />)
    expect(initialMockProps.actions.listJoarkFiles).toHaveBeenCalledWith(initialMockProps.aktoerId)
  })

  it('UseEffect: when new file is available, load it', () => {
    const mockFile = {
      name: 'file.txt',
      dokumentInfoId: '123',
      journalpostId: '123',
      variant: 'foo',
      content: {
        base64: '1232341234234'
      }
    }
    wrapper.setProps({ file: mockFile })
    wrapper.update()
    expect(initialMockProps.onFilesChange).toHaveBeenCalledWith([mockFile])
  })

  it('UseEffect: when new preview file is available, trigger it', () => {
    const mockFile = {
      name: 'file.txt',
      dokumentInfoId: '123',
      journalpostId: '123',
      variant: 'foo',
      content: {
        base64: '1232341234234'
      }
    }
    wrapper.setProps({ previewFile: mockFile })
    wrapper.update()
    expect(initialMockProps.onPreviewFile).toHaveBeenCalledWith(mockFile)
  })

  it('Calls onFilesChange when selecting a file', () => {
    (initialMockProps.onFilesChange as jest.Mock).mockReset()
    wrapper.find('#c-tableSorter__row-checkbox-id-view-1-4-ARKIV').hostNodes().simulate('change', { target: { checked: true } })
    const expectedFile = files[0]
    expectedFile.variant = {
      variantformat: 'ARKIV',
      filnavn: '23534345.pdf'
    }
    expect(initialMockProps.onFilesChange).toHaveBeenCalledWith([{
      date: new Date('2018-12-27T13:42:24.000Z'),
      dokumentInfoId: '4',
      journalpostId: '1',
      key: 'view-1-4-ARKIV',
      label: 'ARKIV (23534345.pdf)',
      name: 'blue.pdf',
      selected: true,
      tema: 'foo',
      variant: {
        filnavn: '23534345.pdf',
        variantformat: 'ARKIV'
      }
    }])
  })

  it('Calls onPreviewItem', () => {
    (initialMockProps.actions.getPreviewJoarkFile as jest.Mock).mockReset()
    wrapper.find('#c-tablesorter__preview-button-1-4-ARKIV__23534345_pdf_').hostNodes().first().simulate('click')
    expect(initialMockProps.actions.getPreviewJoarkFile).toHaveBeenCalledWith(expect.objectContaining({
      date: expect.any(Date),
      dokumentInfoId: '4',
      journalpostId: '1',
      key: 'view-1-4-ARKIV',
      label: 'ARKIV (23534345.pdf)',
      name: 'blue.pdf',
      selected: false,
      tema: 'foo',
      variant: {
        filnavn: '23534345.pdf',
        variantformat: 'ARKIV'
      }
    }))
  })
})
