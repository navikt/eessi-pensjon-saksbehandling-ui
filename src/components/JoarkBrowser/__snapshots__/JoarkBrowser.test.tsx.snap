// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/JoarkBrowser/JoarkBrowser Render: match snapshot 1`] = `
.c1 {
  display: block;
  padding: 1rem 1rem 1rem 1rem;
  border-radius: 4px;
  position: relative;
  -webkit-box-flex: 0;
  -webkit-flex-grow: 0;
  -ms-flex-positive: 0;
  flex-grow: 0;
  overflow: inherit;
  max-height: 100%;
  margin-bottom: 0;
  z-index: 1010;
  color: #3E3832;
  background-color: #FFFFFF;
}

ui-fss.g27[id="c0"] {
  content: "c1,";
}

.c3 {
  display: block !important;
}

.c3 * {
  font-size: inherit;
  line-height: inherit;
}

.c3.compact td,
.c3.compact th {
  padding: 0.35rem !important;
}

.c3.compact th a {
  margin: 0rem !important;
  padding: 0rem !important;
  padding-right: 0.7rem !important;
}

.c3.compact th a:before,
.c3.compact th a:after {
  right: 0rem !important;
}

.c3.compact .c-tableSorter__filter .skjemaelement__input {
  padding: 0.25rem !important;
}

.c3 tbody.striped tr:nth-child(odd) {
  background: #FFFFFF;
}

.c3 tbody.striped tr:nth-child(even) {
  background: #E9E7E7;
}

.c3__subcell {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding: 0.25rem 0.5rem 0.25rem 0.5rem;
}

.c3___seefilters-icon {
  cursor: pointer;
}

.c3__pagination {
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
}

.c3 .header {
  cursor: pointer;
}

.c3 tr.slideAnimate {
  opacity: 0;
  -webkit-transform: translateX(-20px);
  -ms-transform: translateX(-20px);
  transform: translateX(-20px);
  -webkit-animation: kvcTlm 0.2s forwards;
  animation: kvcTlm 0.2s forwards;
}

.c3 .tabell__tr--disabled td {
  background: #B7B1A9 !important;
  color: #FFFFFF !important;
}

.c3 .tabell__tr--disabled td * {
  color: #FFFFFF !important;
}

ui-fss.g53[id="c2"] {
  content: "c3,";
}

.c5 {
  position: relative;
}

ui-fss.g54[id="c4"] {
  content: "c5,";
}

.c7 {
  width: 100%;
}

ui-fss.g56[id="c6"] {
  content: "c7,";
}

.c9 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

ui-fss.g57[id="c8"] {
  content: "c9,";
}

<JoarkBrowser
  existingItems={Array []}
  highContrast={false}
  mode="view"
  onPreviewFile={[MockFunction]}
  onRowSelectChange={[MockFunction]}
  onRowViewDelete={[MockFunction]}
  tableId="test-table-id"
>
  <div
    data-test-id="c-joarkBrowser"
  >
    <Modal
      highContrast={false}
      onModalClose={[Function]}
    >
      <Se
        highContrast={false}
      >
        <Component
          theme={
            Object {
              "alternative-background-color": "#E9E7E7",
              "black": "#000000",
              "border-radius-base": "4px",
              "border-radius-large": "6px",
              "border-radius-small": "2px",
              "container-desktop": "(940px + @grid-gutter-width)",
              "container-large-desktop": "(1140px + @grid-gutter-width)",
              "container-lg": "(1140px + @grid-gutter-width)",
              "container-md": "(940px + @grid-gutter-width)",
              "container-sm": "(720px + @grid-gutter-width)",
              "container-tablet": "(720px + @grid-gutter-width)",
              "fokusFarge": "#254a6c",
              "grayBackground": "#E9E7E7",
              "grayIcon": "#B7B1A9",
              "grayInactive": "#78706A",
              "grayModia": "#333333",
              "grid-columns": "12",
              "grid-float-breakpoint": "768px",
              "grid-float-breakpoint-max": "(@grid-float-breakpoint - 1)",
              "grid-gutter-width": "16px",
              "inverted-font-color": "#FFFFFF",
              "line-height-large": "1.3333333",
              "line-height-small": "1.5",
              "main-active-color": "#a3dff3",
              "main-background-color": "#FFFFFF",
              "main-border-color": "#78706A",
              "main-border-radius": "4px",
              "main-disabled-color": "#B7B1A9",
              "main-error-color": "#BA3A26",
              "main-focus-color": "#254a6c",
              "main-font-color": "#3E3832",
              "main-hover-color": "#e0f4fb",
              "main-interactive-color": "#0067C5",
              "main-success-color": "#06893A",
              "main-warning-color": "#FF9100",
              "navBla": "#0067C5",
              "navBlaDarken20": "#0c5da7",
              "navBlaDarken40": "#18548a",
              "navBlaDarken60": "#254a6c",
              "navBlaDarken80": "#31414f",
              "navBlaLighten20": "#3385d0",
              "navBlaLighten40": "#66a3dc",
              "navBlaLighten60": "#99c2e7",
              "navBlaLighten80": "#cce0f3",
              "navDypBla": "#005B82",
              "navDypBlaDarken20": "#0c5472",
              "navDypBlaDarken40": "#184d62",
              "navDypBlaDarken60": "#254652",
              "navDypBlaDarken80": "#313f42",
              "navDypBlaLighten20": "#337b9b",
              "navDypBlaLighten40": "#669cb4",
              "navDypBlaLighten60": "#99bdcd",
              "navDypBlaLighten80": "#ccdee6",
              "navGra20": "#C6C2BF",
              "navGra40": "#B7B1A9",
              "navGra60": "#78706A",
              "navGra80": "#59514B",
              "navGronn": "#06893A",
              "navGronnDarken20": "#117838",
              "navGronnDarken40": "#1c6836",
              "navGronnDarken60": "#275835",
              "navGronnDarken80": "#324833",
              "navGronnLighten20": "#37a061",
              "navGronnLighten40": "#69b888",
              "navGronnLighten60": "#9bcfb0",
              "navGronnLighten80": "#cde7d7",
              "navLilla": "#634689",
              "navLillaDarken20": "#5b4377",
              "navLillaDarken40": "#544066",
              "navLillaDarken60": "#4c3d54",
              "navLillaDarken80": "#453a43",
              "navLillaLighten20": "#826ba0",
              "navLillaLighten40": "#a190b8",
              "navLillaLighten60": "#c0b5cf",
              "navLillaLighten80": "#dfdae7",
              "navLimeGronn": "#A2AD00",
              "navLimeGronnDarken20": "#8e950a",
              "navLimeGronnDarken40": "#7a7e14",
              "navLimeGronnDarken60": "#66661e",
              "navLimeGronnDarken80": "#524f28",
              "navLimeGronnLighten20": "#b4bd33",
              "navLimeGronnLighten40": "#c7cd66",
              "navLimeGronnLighten60": "#d9de99",
              "navLimeGronnLighten80": "#eceecc",
              "navLysBla": "#66CBEC",
              "navLysBlaDarken20": "#5eadc6",
              "navLysBlaDarken40": "#5690a1",
              "navLysBlaDarken60": "#4e727c",
              "navLysBlaDarken80": "#465557",
              "navLysBlaLighten20": "#84d5ef",
              "navLysBlaLighten40": "#a3dff3",
              "navLysBlaLighten60": "#c1eaf7",
              "navLysBlaLighten80": "#e0f4fb",
              "navLysGra": "#E9E7E7",
              "navMorkGra": "#3E3832",
              "navOransje": "#FF9100",
              "navOransjeDarken20": "#d87f0a",
              "navOransjeDarken40": "#b16d14",
              "navOransjeDarken60": "#8b5b1e",
              "navOransjeDarken80": "#644928",
              "navOransjeLighten20": "#ffa733",
              "navOransjeLighten40": "#ffbd66",
              "navOransjeLighten60": "#ffd399",
              "navOransjeLighten80": "#ffe9cc",
              "navRod": "#C30000",
              "navRodDarken20": "#a80b0a",
              "navRodDarken40": "#8d1614",
              "navRodDarken60": "#73211e",
              "navRodDarken80": "#582c28",
              "navRodLighten20": "#cf3333",
              "navRodLighten40": "#db6666",
              "navRodLighten60": "#e79999",
              "navRodLighten80": "#f3cccc",
              "orangeFocus": "#FFBD66",
              "orangeFocusDarken20": "#d8a25b",
              "orangeFocusDarken40": "#b18751",
              "orangeFocusDarken60": "#8b6d46",
              "orangeFocusDarken80": "#64523c",
              "orangeFocusLighten20": "#ffca84",
              "orangeFocusLighten40": "#ffd7a3",
              "orangeFocusLighten60": "#ffe4c1",
              "orangeFocusLighten80": "#fff1e0",
              "padding-base-horizontal": "12px",
              "padding-base-vertical": "6px",
              "padding-large-horizontal": "16px",
              "padding-large-vertical": "10px",
              "padding-small-horizontal": "10px",
              "padding-small-vertical": "5px",
              "padding-xs-horizontal": "5px",
              "padding-xs-vertical": "1px",
              "pinkErrorBg": "#F3E3E3",
              "redError": "#BA3A26",
              "redErrorDarken20": "#a13928",
              "redErrorDarken40": "#88392a",
              "redErrorDarken60": "#6f382d",
              "redErrorDarken80": "#56382f",
              "redErrorLighten20": "#c76151",
              "redErrorLighten40": "#d5887c",
              "redErrorLighten60": "#e3b0a8",
              "redErrorLighten80": "#f1d7d3",
              "screen-desktop": "992px",
              "screen-lg": "1200px",
              "screen-lg-desktop": "1200px",
              "screen-lg-min": "1200px",
              "screen-md": "992px",
              "screen-md-max": "(@screen-lg-min - 1)",
              "screen-md-min": "992px",
              "screen-sm": "768px",
              "screen-sm-max": "(@screen-md-min - 1)",
              "screen-sm-min": "768px",
              "screen-tablet": "768px",
              "screen-xs-max": "(@screen-sm-min - 1)",
              "type": "theme",
              "white": "#FFFFFF",
            }
          }
        >
          <Memo(l) />
          <Styled(Modal)
            isOpen={false}
            onRequestClose={[Function]}
            overlayClassName="modal__overlay"
          >
            <Modal
              ariaHideApp={true}
              bodyOpenClassName="ReactModal__Body--open"
              className="c0 c1"
              closeTimeoutMS={0}
              contentElement={[Function]}
              isOpen={false}
              onRequestClose={[Function]}
              overlayClassName="modal__overlay"
              overlayElement={[Function]}
              parentSelector={[Function]}
              portalClassName="ReactModalPortal"
              preventScroll={false}
              role="dialog"
              shouldCloseOnEsc={true}
              shouldCloseOnOverlayClick={true}
              shouldFocusAfterRender={true}
              shouldReturnFocusAfterClose={true}
            >
              <Portal
                containerInfo={
                  <div
                    class="ReactModalPortal"
                  />
                }
              >
                <ModalPortal
                  ariaHideApp={true}
                  bodyOpenClassName="ReactModal__Body--open"
                  className="c0 c1"
                  closeTimeoutMS={0}
                  contentElement={[Function]}
                  defaultStyles={
                    Object {
                      "content": Object {
                        "WebkitOverflowScrolling": "touch",
                        "background": "#fff",
                        "border": "1px solid #ccc",
                        "borderRadius": "4px",
                        "bottom": "40px",
                        "left": "40px",
                        "outline": "none",
                        "overflow": "auto",
                        "padding": "20px",
                        "position": "absolute",
                        "right": "40px",
                        "top": "40px",
                      },
                      "overlay": Object {
                        "backgroundColor": "rgba(255, 255, 255, 0.75)",
                        "bottom": 0,
                        "left": 0,
                        "position": "fixed",
                        "right": 0,
                        "top": 0,
                      },
                    }
                  }
                  isOpen={false}
                  onRequestClose={[Function]}
                  overlayClassName="modal__overlay"
                  overlayElement={[Function]}
                  parentSelector={[Function]}
                  portalClassName="ReactModalPortal"
                  preventScroll={false}
                  role="dialog"
                  shouldCloseOnEsc={true}
                  shouldCloseOnOverlayClick={true}
                  shouldFocusAfterRender={true}
                  shouldReturnFocusAfterClose={true}
                  style={
                    Object {
                      "content": Object {},
                      "overlay": Object {},
                    }
                  }
                />
              </Portal>
            </Modal>
          </Styled(Modal)>
        </Component>
      </Se>
    </Modal>
    <Dt
      columns={
        Array [
          Object {
            "id": "tema",
            "label": "ui:tema",
            "renderCell": [Function],
            "type": "string",
          },
          Object {
            "id": "title",
            "label": "ui:title",
            "type": "string",
          },
          Object {
            "dateFormat": "DD.MM.YYYY",
            "id": "date",
            "label": "ui:date",
            "type": "date",
          },
          Object {
            "id": "buttons",
            "label": "",
            "renderCell": [Function],
            "type": "object",
          },
        ]
      }
      compact={true}
      context={
        Object {
          "clickedPreviewItem": undefined,
          "existingItems": Array [],
          "loadingJoarkPreviewFile": false,
          "mode": "view",
          "previewFile": undefined,
        }
      }
      highContrast={false}
      id="joarkbrowser-test-table-id"
      items={Array []}
      itemsPerPage={30}
      labels={
        Object {
          "type": "ui:attachments",
        }
      }
      loading={false}
      onRowSelectChange={[MockFunction]}
      searchable={false}
      selectable={false}
      sortable={false}
      summary={true}
    >
      <Component
        theme={
          Object {
            "alternative-background-color": "#E9E7E7",
            "black": "#000000",
            "border-radius-base": "4px",
            "border-radius-large": "6px",
            "border-radius-small": "2px",
            "container-desktop": "(940px + @grid-gutter-width)",
            "container-large-desktop": "(1140px + @grid-gutter-width)",
            "container-lg": "(1140px + @grid-gutter-width)",
            "container-md": "(940px + @grid-gutter-width)",
            "container-sm": "(720px + @grid-gutter-width)",
            "container-tablet": "(720px + @grid-gutter-width)",
            "fokusFarge": "#254a6c",
            "grayBackground": "#E9E7E7",
            "grayIcon": "#B7B1A9",
            "grayInactive": "#78706A",
            "grayModia": "#333333",
            "grid-columns": "12",
            "grid-float-breakpoint": "768px",
            "grid-float-breakpoint-max": "(@grid-float-breakpoint - 1)",
            "grid-gutter-width": "16px",
            "inverted-font-color": "#FFFFFF",
            "line-height-large": "1.3333333",
            "line-height-small": "1.5",
            "main-active-color": "#a3dff3",
            "main-background-color": "#FFFFFF",
            "main-border-color": "#78706A",
            "main-border-radius": "4px",
            "main-disabled-color": "#B7B1A9",
            "main-error-color": "#BA3A26",
            "main-focus-color": "#254a6c",
            "main-font-color": "#3E3832",
            "main-hover-color": "#e0f4fb",
            "main-interactive-color": "#0067C5",
            "main-success-color": "#06893A",
            "main-warning-color": "#FF9100",
            "navBla": "#0067C5",
            "navBlaDarken20": "#0c5da7",
            "navBlaDarken40": "#18548a",
            "navBlaDarken60": "#254a6c",
            "navBlaDarken80": "#31414f",
            "navBlaLighten20": "#3385d0",
            "navBlaLighten40": "#66a3dc",
            "navBlaLighten60": "#99c2e7",
            "navBlaLighten80": "#cce0f3",
            "navDypBla": "#005B82",
            "navDypBlaDarken20": "#0c5472",
            "navDypBlaDarken40": "#184d62",
            "navDypBlaDarken60": "#254652",
            "navDypBlaDarken80": "#313f42",
            "navDypBlaLighten20": "#337b9b",
            "navDypBlaLighten40": "#669cb4",
            "navDypBlaLighten60": "#99bdcd",
            "navDypBlaLighten80": "#ccdee6",
            "navGra20": "#C6C2BF",
            "navGra40": "#B7B1A9",
            "navGra60": "#78706A",
            "navGra80": "#59514B",
            "navGronn": "#06893A",
            "navGronnDarken20": "#117838",
            "navGronnDarken40": "#1c6836",
            "navGronnDarken60": "#275835",
            "navGronnDarken80": "#324833",
            "navGronnLighten20": "#37a061",
            "navGronnLighten40": "#69b888",
            "navGronnLighten60": "#9bcfb0",
            "navGronnLighten80": "#cde7d7",
            "navLilla": "#634689",
            "navLillaDarken20": "#5b4377",
            "navLillaDarken40": "#544066",
            "navLillaDarken60": "#4c3d54",
            "navLillaDarken80": "#453a43",
            "navLillaLighten20": "#826ba0",
            "navLillaLighten40": "#a190b8",
            "navLillaLighten60": "#c0b5cf",
            "navLillaLighten80": "#dfdae7",
            "navLimeGronn": "#A2AD00",
            "navLimeGronnDarken20": "#8e950a",
            "navLimeGronnDarken40": "#7a7e14",
            "navLimeGronnDarken60": "#66661e",
            "navLimeGronnDarken80": "#524f28",
            "navLimeGronnLighten20": "#b4bd33",
            "navLimeGronnLighten40": "#c7cd66",
            "navLimeGronnLighten60": "#d9de99",
            "navLimeGronnLighten80": "#eceecc",
            "navLysBla": "#66CBEC",
            "navLysBlaDarken20": "#5eadc6",
            "navLysBlaDarken40": "#5690a1",
            "navLysBlaDarken60": "#4e727c",
            "navLysBlaDarken80": "#465557",
            "navLysBlaLighten20": "#84d5ef",
            "navLysBlaLighten40": "#a3dff3",
            "navLysBlaLighten60": "#c1eaf7",
            "navLysBlaLighten80": "#e0f4fb",
            "navLysGra": "#E9E7E7",
            "navMorkGra": "#3E3832",
            "navOransje": "#FF9100",
            "navOransjeDarken20": "#d87f0a",
            "navOransjeDarken40": "#b16d14",
            "navOransjeDarken60": "#8b5b1e",
            "navOransjeDarken80": "#644928",
            "navOransjeLighten20": "#ffa733",
            "navOransjeLighten40": "#ffbd66",
            "navOransjeLighten60": "#ffd399",
            "navOransjeLighten80": "#ffe9cc",
            "navRod": "#C30000",
            "navRodDarken20": "#a80b0a",
            "navRodDarken40": "#8d1614",
            "navRodDarken60": "#73211e",
            "navRodDarken80": "#582c28",
            "navRodLighten20": "#cf3333",
            "navRodLighten40": "#db6666",
            "navRodLighten60": "#e79999",
            "navRodLighten80": "#f3cccc",
            "orangeFocus": "#FFBD66",
            "orangeFocusDarken20": "#d8a25b",
            "orangeFocusDarken40": "#b18751",
            "orangeFocusDarken60": "#8b6d46",
            "orangeFocusDarken80": "#64523c",
            "orangeFocusLighten20": "#ffca84",
            "orangeFocusLighten40": "#ffd7a3",
            "orangeFocusLighten60": "#ffe4c1",
            "orangeFocusLighten80": "#fff1e0",
            "padding-base-horizontal": "12px",
            "padding-base-vertical": "6px",
            "padding-large-horizontal": "16px",
            "padding-large-vertical": "10px",
            "padding-small-horizontal": "10px",
            "padding-small-vertical": "5px",
            "padding-xs-horizontal": "5px",
            "padding-xs-vertical": "1px",
            "pinkErrorBg": "#F3E3E3",
            "redError": "#BA3A26",
            "redErrorDarken20": "#a13928",
            "redErrorDarken40": "#88392a",
            "redErrorDarken60": "#6f382d",
            "redErrorDarken80": "#56382f",
            "redErrorLighten20": "#c76151",
            "redErrorLighten40": "#d5887c",
            "redErrorLighten60": "#e3b0a8",
            "redErrorLighten80": "#f1d7d3",
            "screen-desktop": "992px",
            "screen-lg": "1200px",
            "screen-lg-desktop": "1200px",
            "screen-lg-min": "1200px",
            "screen-md": "992px",
            "screen-md-max": "(@screen-lg-min - 1)",
            "screen-md-min": "992px",
            "screen-sm": "768px",
            "screen-sm-max": "(@screen-md-min - 1)",
            "screen-sm-min": "768px",
            "screen-tablet": "768px",
            "screen-xs-max": "(@screen-sm-min - 1)",
            "type": "theme",
            "white": "#FFFFFF",
          }
        }
      >
        <styled.div
          className="tabell compact"
        >
          <div
            className="c2 c3 tabell compact"
          >
            <styled.div>
              <div
                className="c4 c5"
              >
                <styled.table
                  cellSpacing="0"
                  className="c-tableSorter__table"
                >
                  <table
                    cellSpacing="0"
                    className="c6 c7 c-tableSorter__table"
                  >
                    <thead>
                      <tr
                        className="c-tableSorter__header"
                      >
                        <th
                          style={
                            Object {
                              "width": 1,
                            }
                          }
                        >
                          <div
                            style={
                              Object {
                                "alignItems": "center",
                                "display": "flex",
                              }
                            }
                          />
                        </th>
                        <th
                          aria-sort="none"
                          className="header "
                          key="tema"
                          role="columnheader"
                        >
                          ui:tema
                        </th>
                        <th
                          aria-sort="none"
                          className="header "
                          key="title"
                          role="columnheader"
                        >
                          ui:title
                        </th>
                        <th
                          aria-sort="none"
                          className="header "
                          key="date"
                          role="columnheader"
                        >
                          ui:date
                        </th>
                        <th
                          aria-sort="none"
                          className="header"
                          key="buttons"
                          role="columnheader"
                        />
                      </tr>
                    </thead>
                    <tbody
                      className="striped"
                    />
                  </table>
                </styled.table>
                <styled.div>
                  <div
                    className="c8 c9"
                  >
                    <div />
                    <i>
                      <s
                        type="normaltekst"
                      >
                        <p
                          className="typo-normal"
                        >
                          Ingen ui:attachments vist
                        </p>
                      </s>
                    </i>
                    <g
                      currentPage={1}
                      highContrast={false}
                      itemsPerPage={30}
                      numberOfItems={0}
                      onChange={[Function]}
                    />
                  </div>
                </styled.div>
              </div>
            </styled.div>
          </div>
        </styled.div>
      </Component>
    </Dt>
  </div>
</JoarkBrowser>
`;
